routes/dao_coin_exchange.go

```diff
@@ -581,3 +581,117 @@ func getNumDigits(val *big.Int) int {
+	sellingCoinPKID := &lib.ZeroPKID
+
+	// Calculate current balance for transactor.
+	transactorSellingBalanceBaseUnits := uint256.NewInt()
+	if requestData.SellingDAOCoinCreatorPublicKeyBase58Check == DESOCoinIdentifierString {
+		// Get $DESO balance nanos.
+		desoBalanceNanos, err := utxoView.GetDeSoBalanceNanosForPublicKey(transactorPublicKey)
+		if err != nil {
+			return errors.Errorf("Error getting transactor DESO balance: %v", err)
+		}
+		transactorSellingBalanceBaseUnits = uint256.NewInt().SetUint64(desoBalanceNanos)
+	} else {
+		// Get selling coin PKID and public key from public key base58 check.
+		sellingCoinPKID, err = fes.getPKIDFromPublicKeyBase58Check(
+			utxoView, requestData.SellingDAOCoinCreatorPublicKeyBase58Check)
+		if err != nil {
+			return errors.Errorf("Invalid SellingDAOCoinCreatorPublicKeyBase58Check: %v", err)
+		}
+		sellingPublicKey, _, err := lib.Base58CheckDecode(requestData.SellingDAOCoinCreatorPublicKeyBase58Check)
+		if err != nil {
+			return errors.Errorf("Error decoding selling public key: %v", err)
+		}
+
+		// Get DAO coin balance base units.
+		balanceEntry, _, _ := utxoView.GetBalanceEntryForHODLerPubKeyAndCreatorPubKey(transactorPublicKey, sellingPublicKey, true)
+		if balanceEntry == nil {
+			return errors.New("Error getting transactor DAO coin balance not found")
+		}
+		transactorSellingBalanceBaseUnits = &balanceEntry.BalanceNanos
+	}
+
+	// Get open orders for this transactor
+	orders, err := utxoView.GetAllDAOCoinLimitOrdersForThisTransactor(transactorPKID)
+	if err != nil {
+		return errors.Errorf("Error getting limit orders: %v", err)
+	}
+
+	// Calculate total selling quantity for current order.
+	totalSellingBaseUnits := uint256.NewInt()
+	if requestData.OperationType == DAOCoinLimitOrderOperationTypeStringASK {
+		totalSellingBaseUnits = quantityToFillInBaseUnits
+	} else if requestData.OperationType == DAOCoinLimitOrderOperationTypeStringBID {
+		totalSellingBaseUnits, err = lib.ComputeBaseUnitsToSellUint256(
+			scaledExchangeRateCoinsToSellPerCoinToBuy, quantityToFillInBaseUnits)
+		if err != nil {
+			return errors.Errorf("Error calculating new order selling quantity: %v", err)
+		}
+	} else {
+		return errors.Errorf("Invalid operation type: %s", requestData.OperationType)
+	}
+
+	// Add total selling quantity for existing/open orders.
+	for _, order := range orders {
+		if buyingCoinPKID.Eq(order.BuyingDAOCoinCreatorPKID) &&
+			sellingCoinPKID.Eq(order.SellingDAOCoinCreatorPKID) {
+			// Calculate selling quantity.
+			orderSellingBaseUnits, err := order.BaseUnitsToSellUint256()
+			if err != nil {
+				return errors.Errorf("Error calculating open order selling quantity: %v", err)
+			}
+
+			// Sum selling quantity.
+			totalSellingBaseUnits, err = lib.SafeUint256().Add(totalSellingBaseUnits, orderSellingBaseUnits)
+			if err != nil {
+				return errors.Errorf("Error adding open order selling quantity: %v", err)
+			}
```

Comment:
May as well make the `transactorSellingBalanceBaseUnits.Lt(totalSellingBaseUnits)` check here and exiting early if true